tests/test_story.py

```diff
@@ -32,8 +31,107 @@ def do(I):
     assert str(exc_info.value) == "Story should have at least one step defined"
 
 
-def test_story_representation(x):
+def test_story_with_sync_and_async_steps():
+    """Deny building stories with both sync and async steps"""
+
+    with pytest.raises(StoryDefinitionError) as exc_info:
+        class Action(object):
+            @story
+            def do(I):
+                I.sync
+                I.async_
+
+            def sync(self, ctx):
+                pass
+
+            async def async_(self, ctx):
+                pass
+
+        Action().do.run()
+
+    assert str(exc_info.value) == """
+Action.async_ is a coroutine but a function was expected.
+Story method: Action.do
+You can not use function and coroutine methods together.
+""".strip()
+
+
+def test_story_with_async_and_sync_steps():
+    """Deny building stories with both sync and async steps"""
 
+    with pytest.raises(StoryDefinitionError) as exc_info:
+        class Action(object):
+            @story
+            def do(I):
+                I.async_
+                I.sync
+
+            def sync(self, ctx):
+                pass
+
+            async def async_(self, ctx):
+                pass
+
+        Action().do.run()
+
+    assert str(exc_info.value) == """
+Action.sync is a function but a coroutine was expected.
+Story method: Action.do
+You can not use function and coroutine methods together.
+""".strip()
+
+
+def test_sync_story_with_async_substory():
+    with pytest.raises(StoryDefinitionError) as exc_info:
+        class Action(object):
+            @story
+            def do(I):
+                I.sync
+                I.do2
+
+            def sync(self, ctx):
+                pass
+
+            @story
+            def do2(I):
+                I.async_
+
+            async def async_(self, ctx):
+                pass
+
+        Action().do.run()
+
+    assert str(exc_info.value) == """
+Action.do expects coroutine methods but Action.do2 consists of function methods.
```

Comment:
Actually `Action.do` expects a function since it is a sync story.
This is a bug we need to fix.